src/theories/Category.jl

```diff
@@ -121,3 +122,106 @@ end
 function show_latex(io::IO, expr::Hom2Expr{:compose2}; kw...)
   Syntax.show_latex_infix(io, expr, "*"; kw...)
 end
+
+
+###########
+# Double Category
+###########
+
+""" Theory of (strict) *double categories*
+"""
+@theory DoubleCategory{Ob,HomV,HomH,Hom2} begin
+  # """ Object in a category """
+  Ob::TYPE
+  """ Horizontal Morphism in a double category """
+  HomH(dom::Ob, codom::Ob)::TYPE
+  """ Vertical Morphism in a double category """
+  HomV(dom::Ob, codom::Ob)::TYPE
+  """ 2-cell in a double category """
+  Hom2(top::HomH(A,B),
+       bottom::HomH(C,D),
+       left::HomV(A,C),
+       right::HomV(B,D))::TYPE ⊣ (A::Ob, B::Ob, C::Ob, D::Ob)
+  @op begin
+    (→) := HomH
+    (↓) := HomV
+    (⇒) := Hom2
+    (⋅) := composeH
+    (⋆) := composeV
+  end
+
+  idH(A::Ob)::(A → A) ⊣ (A::Ob)
+  idV(A::Ob)::(A ↓ A) ⊣ (A::Ob)
+  composeH(f::(A → B), g::(B → C))::(A → C) ⊣ (A::Ob, B::Ob, C::Ob)
+  composeV(f::(A ↓ B), g::(B ↓ C))::(A ↓ C) ⊣ (A::Ob, B::Ob, C::Ob)
+
+  # Category axioms for Horizontal morphisms
+  ((f ⋅ g) ⋅ h == f ⋅ (g ⋅ h)
+    ⊣ (A::Ob, B::Ob, C::Ob, D::Ob, f::(A → B), g::(B → C), h::(C → D)))
+  f ⋅ idH(B) == f ⊣ (A::Ob, B::Ob, f::(A → B))
+  idH(A) ⋅ f == f ⊣ (A::Ob, B::Ob, f::(A → B))
+
+  # Category axioms for Vertical morphisms
+  ((f ⋅ g) ⋅ h == f ⋅ (g ⋅ h)
+    ⊣ (A::Ob, B::Ob, C::Ob, D::Ob, f::(A ↓ B), g::(B ↓ C), h::(C ↓ D)))
+  f ⋅ idV(B) == f ⊣ (A::Ob, B::Ob, f::(A ↓ B))
+  idV(A) ⋅ f == f ⊣ (A::Ob, B::Ob, f::(A ↓ B))
+
+  # identity two cell on 1 object
+  id2(X::Ob)::Hom2(X→X, X→X, X↓X, X↓X)          ⊣ (X::Ob)
+  id2(X) == id2(idH(X), idH(X), idV(X), idV(X)) ⊣ (X::Ob)
+  # identity two cell from a HomH
+  id2(f::(X→Y))::Hom2(X→Y, X→Y, X↓X, Y↓Y) ⊣ (X::Ob, Y::Ob)
+  id2(f) == Hom2(f, f, idV(X), idV(Y))    ⊣ (X::Ob, Y::Ob, f::(X→Y))
+  # identity two cell from a HomV
+  id2(f::(X↓Y))::Hom2(X→X, Y→Y, X↓Y, X↓Y) ⊣ (X::Ob, Y::Ob)
+  id2(f) == Hom2(idH(X), idH(Y), f, f)    ⊣ (X::Ob, Y::Ob, f::(X→Y))
+
```

Comment:
No associativity/ unitality for 2-cell composition? This GAT is already pretty brutal, so I can understand why, but maybe add a comment? Though, we have to think a bit about weakness, what is the Catlab policy on weakness?

Reply:
Yeah I didn't know what the right answer was for that so I left it out. What are the important issues about weakness that you can see?

Reply:
So far weakness is not an issue that we've had to address in Catlab. All the theories for monoidal categories are strict. IMO, we should not worry about this until it actually becomes a problem. Typically we strictify everything, e.g. working with the skeleton of FinSet rather than Set.